tests/end-to-end/backup-restore-test/backupe2e/deployment.go

```diff
@@ -194,8 +194,8 @@ func (d deploymentTest) waitForPodDeployment(namespace string, replicas int32, w
 			if err != nil {
 				return err
 			}
-			if deployment.Status.AvailableReplicas != replicas {
-				return fmt.Errorf("Number of pods %v different from available replicas %v", len(pods.Items), replicas)
+			if deployment.Status.ReadyReplicas != replicas {
+				break
```

Comment:
Can we merge https://github.com/kyma-project/kyma/blob/dc6a72059b357e2a2f40efa3f06a8e933e2b44fd/tests/end-to-end/backup-restore-test/backupe2e/deployment.go#L190 with this?